Task 1 По данным опроса, 75% работников ресторанов утверждают, что испытывают на работе существенный стресс, оказывающий негативное влияние на их личную жизнь. Крупная ресторанная сеть опрашивает 100 своих работников, чтобы выяснить, отличается ли уровень стресса работников в их ресторанах от среднего. 67 из 100 работников отметили высокий уровень стресса.

Посчитайте достигаемый уровень значимости, округлите ответ до четырёх знаков после десятичной точки.

Переформулируем вопрос в терминах проверки гипотез:"отличается ли уровень стресса сотрудников ресторана от среднего?" 

Имеем дело с выборкой, состоящей из 0 и 1, то есть, из распределения Бернулли  Ber(p) . 

Гипотеза H0- верно ли сети ресторанов распределение ответов сотрудников -биномиальное распределение  Bin(n,p) c p=0.75 ? 

Гипотеза H1: p не равно 0.75. Используем друстороннюю альтернативу, учитываем как правый так и левый конец, так как интересует отклонение доли от среднего.

In [2]:
import numpy as np
from scipy import stats

In [3]:
%pylab inline

Populating the interactive namespace from numpy and matplotlib


гипотеза  H : Джеймс Бонд предпочитает какой-то определённый вид мартини, при этом он не выбираем какой именно. При расчете будем учитывать как правый, так и левый конец.


In [114]:
stats.binom_test(67, 100, 0.75, alternative = 'two-sided')

0.082222588913866079

Таsk 2 Представим теперь, что в другой ресторанной сети только 22 из 50 работников испытывают существенный стресс. Гипотеза о том, что 22/50 соответствует 75% по всей популяции, методом, который вы использовали в предыдущей задаче, отвергается. Чем это может объясняться? Выберите все возможные варианты.

In [5]:
stats.binom_test(44, 100, 0.75, alternative = 'two-sided')

5.1571289311031216e-11

Условия работы в этой сети лучше, чем в среднем

Task 3 The Wage Tract — заповедник в округе Тома, Джорджия, США, деревья в котором не затронуты деятельностью человека со времён первых поселенцев. Для участка заповедника размером 200х200 м имеется информация о координатах сосен (sn — координата в направлении север-юг, we — в направлении запад-восток, обе от 0 до 200).

pines.txt
Проверим, можно ли пространственное распределение сосен считать равномерным, или они растут кластерами.

Загрузите данные, поделите участок на 5х5 одинаковых квадратов размера 40x40 м, посчитайте количество сосен в каждом квадрате (чтобы получить такой же результат, как у нас, используйте функцию scipy.stats.binned_statistic_2d).

scipy.stats.binned_statistic_2d(x, y, values, statistic='mean', bins=10, range=None, expand_binnumbers=False)[source]
Compute a bidimensional binned statistic for one or more sets of data.

bins : int or [int, int] or array_like or [array, array], optional
Если сосны действительно растут равномерно, какое среднее ожидаемое количество сосен в каждом квадрате? В правильном ответе два знака после десятичной точки.


In [80]:
import pandas as pd
data = pd.read_csv("./pines.csv")
data.head(5)

sn\twe
0     200\t8.8
1    199.3\t10
2  193.6\t22.4
3  167.7\t35.6
4  183.9\t45.4

In [81]:
data = pd.read_csv('./pines.csv', header = None, sep = '\t')
data.columns = ['sn', 'we']
data.head()

sn    we
0     sn    we
1    200   8.8
2  199.3    10
3  193.6  22.4
4  167.7  35.6

In [82]:
data.shape

(585, 2)

In [83]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 585 entries, 0 to 584
Data columns (total 2 columns):
sn    585 non-null object
we    585 non-null object
dtypes: object(2)
memory usage: 9.2+ KB


In [84]:
data.describe()

sn     we
count     585    585
unique    518    494
top     101.5  127.4
freq        3      5

In [101]:
s=data.we
s0=data.sn

In [102]:
s=s.convert_objects(convert_numeric=True)
s0=s0.convert_objects(convert_numeric=True)

//anaconda/lib/python3.5/site-packages/ipykernel/__main__.py:1: FutureWarning: convert_objects is deprecated.  Use the data-type specific converters pd.to_datetime, pd.to_timedelta and pd.to_numeric.
  if __name__ == '__main__':
//anaconda/lib/python3.5/site-packages/ipykernel/__main__.py:2: FutureWarning: convert_objects is deprecated.  Use the data-type specific converters pd.to_datetime, pd.to_timedelta and pd.to_numeric.
  from ipykernel import kernelapp as app


In [72]:
data.we.reshape(-1,1)

array([['we'],
       ['8.8'],
       ['10'],
       ['22.4'],
       ['35.6'],
       ['45.4'],
       ['47.2'],
       ['48.8'],
       ['42.1'],
       ['29'],
       ['33.6'],
       ['35.8'],
       ['38.7'],
       ['42.8'],
       ['36.2'],
       ['45.8'],
       ['25.4'],
       ['25.4'],
       ['28.3'],
       ['36.9'],
       ['14.3'],
       ['13.1'],
       ['8.1'],
       ['9.3'],
       ['9.1'],
       ['14.7'],
       ['29.7'],
       ['45.8'],
       ['44.2'],
       ['49.4'],
       ['49.6'],
       ['10.8'],
       ['0'],
       ['5.4'],
       ['5.5'],
       ['3.9'],
       ['8.5'],
       ['11.3'],
       ['11'],
       ['9.2'],
       ['12.6'],
       ['21.4'],
       ['39.8'],
       ['18.2'],
       ['46.9'],
       ['46.9'],
       ['47.9'],
       ['50'],
       ['47.9'],
       ['47.9'],
       ['50.7'],
       ['45.2'],
       ['35'],
       ['35.5'],
       ['44.8'],
       ['43.4'],
       ['40.2'],
       ['34.3'],
       ['33.3'],
       ['29.8'],
    

In [107]:
our_bins=[0.0,40,80,120,160,200]

In [108]:
res=stats.binned_statistic_2d(x=s0, y=s, values=None,statistic='count', bins=[our_bins,our_bins])

In [109]:
res.statistic

array([[ 18.,  26.,  29.,  25.,  20.],
       [ 14.,  12.,  22.,  34.,  25.],
       [ 12.,  24.,  15.,  50.,  37.],
       [  8.,  19.,  31.,  50.,   7.],
       [  7.,   8.,  37.,  27.,  27.]])

In [110]:
mean(list(map(np.mean, res.statistic)))

23.359999999999999

4. 
Чтобы сравнить распределение сосен с равномерным, посчитайте значение статистики хи-квадрат для полученных 5х5 квадратов. Округлите ответ до двух знаков после десятичной точки.

In [111]:
stats.chisquare(res.statistic.reshape(-1,1))

Power_divergenceResult(statistic=array([ 150.5890411]), pvalue=array([  2.57466977e-20]))

Получили значение статистики 150.59, p-value практически ноль. Значит, можем отвергнуть гипотезу H0. То есть наши данные не имеют нормальное распределение. 

In [ ]:
5 Насколько велико это значение? Если нулевая гипотеза справедлива, с какой вероятностью его можно было получить случайно?

Нулевое распределение статистики — хи-квадрат с 25−1=24 степенями свободы (поскольку у равномерного распределения, с которым мы сравниваем данные, нет ни одного оцениваемого по выборке параметра, число степеней свободы K−1, где K — количество интервалов).

Посчитайте достигаемый уровень значимости.

Если вы используете функцию scipy.stats.chi2.cdf, в качестве значения параметра df нужно взять 24 (это число степеней свободы); если функцию scipy.stats.chisquare — параметр ddof нужно брать равным 0 (это как раз количество параметров теоретического распределения, оцениваемых по выборке).

Отвергается ли гипотеза равномерности на уровне значимости 0.05?